vocabsieve/main.py

```diff
@@ -702,10 +702,25 @@ def getCurrentWord(self) -> str:
         selected = ""
         for text_field in [self.sentence, self.definition, self.definition2]:
             if text_field.hasFocus():
-                if text := text_field.textCursor().selectedText():
-                    logger.debug(f"Selected text: {text} from text field")
-                    selected = text
+                cursor = text_field.textCursor()
+                if selected := cursor.selectedText():
+                    logger.debug(f"Manually selected text: {selected} from text field {text_field}")
                     break
+                else:
+                    cursor_position = cursor.position()
+
+                    cursor.movePosition(QTextCursor.StartOfWord, QTextCursor.MoveAnchor)
+                    word_start_position = cursor.position()
+
+                    cursor.movePosition(QTextCursor.EndOfWord, QTextCursor.KeepAnchor)
+                    word_end_position = cursor.position()
+
+                    if word_start_position != word_end_position and (word_start_position <= cursor_position <= word_end_position):
+                        selected = text_field.toPlainText()[word_start_position:word_end_position].replace('_', '')
+                        logger.debug(f"Automatic selection of touched word: {selected} from text field {text_field}")
```

Comment:
don't actually print the text_field. If possible print the name of the field, or use enumerate and access the index to get the name back. Moreover, it's not really worth doing  the cursor stuff on the two definition fields because their content is going to be overwritten so you can't assume the same text will be at the same position (or that the word will be present at all. Maybe break up this loop into three segments.